sdk/identity/azure-identity/src/main/java/com/azure/identity/AzureCliCredential.java

```diff
@@ -94,7 +95,7 @@ public AccessToken getTokenSync(TokenRequestContext request) {
             return accessToken;
         } catch (Exception e) {
             LoggingUtil.logTokenError(LOGGER, identityClient.getIdentityClientOptions(), request, e);
-            throw e;
+            throw new CredentialUnavailableException("Azure CLI authentication is not available", e);
```

Comment:
So, we're changing the type thrown here in all cases. Should we add internal methods to know when we're using this as part of chaining to properly handle continuation on a credential not being usable and in other public API cases we retain current functionality? 

Also, do we unwrap this wrapped exception when we hit the end of chaining to throw the appropriate exception to maintain backwards compatibility? Ex, if before I was catching an MSAL exception type when the chain failed to retry in my own application's logic that catch block won't be triggered anymore as the exception type has changed to `CredentialUnavailableException`.

One last thought, do we need more logic here to differentiate between a credential being unavailable and a true exception being thrown, like an `IOException` because of a networking failure?

Reply:
Hm, good point. I'll rework this.

Reply:
I've updated it. We change nothing when you're using the credentials directly. We're not doing the re-throw as you suggested, but I'm okay with that. We're sharing the message out of the original exception as well as linking it as the cause. The stack will be right since we're just wrapping the exception we caught in the same method. Lemme know if you have further concerns. Thanks!

Reply:
I think we have two options.

1. Update the individual credential and if it is called from DAC/chain, we raise `CredentialUnavailableException`
2. We can maintain a list of "dev credentials" in DAC and catch `ClientAuthenticationException` and continue in DAC.

Reply:
We are doing #1.